src/core/syscalls/syscall_handler.rs

```diff
@@ -195,3 +200,41 @@ fn get_syscall_ptr(
         .into_owned();
     Ok(syscall_ptr)
 }
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{bigint, utils::test_utils::vm};
+    use cairo_rs::relocatable;
+    use num_bigint::{BigInt, Sign};
+
+    use super::*;
+
+    #[test]
+    fn read_deploy_syscall_request() {
+        let syscall = BusinessLogicSyscallHandler::new();
+        let mut vm = vm!();
+        vm.add_memory_segment();
```

Comment:
the macro  "add_segments" can be use to avoid code repetition

Reply:
Thanks!
Done!